src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenFunctionPointersTests.cs

```diff
@@ -5174,6 +5175,225 @@ void symbolValidator(ModuleSymbol module)
             }
         }
 
+        [Fact]
+        public void DynamicTypeAttributeInMetadata()
+        {
+            var comp = CompileAndVerifyFunctionPointers(@"
+#pragma warning disable CS0649 // Unassigned field
+unsafe class C
+{
+    public delegate*<dynamic, dynamic, dynamic> F1;
+
+    public delegate*<object, object, dynamic> F2;
+    public delegate*<dynamic, object, object> F3;
+    public delegate*<object, dynamic, object> F4;
+
+    public delegate*<object, object, object> F5;
+
+    public delegate*<object, object, ref dynamic> F6;
+    public delegate*<ref dynamic, object, object> F7;
+    public delegate*<object, ref dynamic, object> F8;
+
+    public delegate*<ref object, ref object, dynamic> F9;
+    public delegate*<dynamic, ref object, ref object> F10;
+    public delegate*<ref object, dynamic, ref object> F11;
+
+    public delegate*<object, ref readonly dynamic> F12;
+    public delegate*<in dynamic, object> F13;
+
+    public delegate*<out dynamic, object> F14;
+}
+", symbolValidator: symbolValidator);
+
+            void symbolValidator(ModuleSymbol module)
+            {
+                var c = module.GlobalNamespace.GetTypeMember("C");
+
+                assertField("F1", "System.Runtime.CompilerServices.DynamicAttribute({false, true, true, true})", "delegate*<dynamic, dynamic, dynamic>");
+
+                assertField("F2", "System.Runtime.CompilerServices.DynamicAttribute({false, true, false, false})", "delegate*<System.Object, System.Object, dynamic>");
```

Comment:
>true, false, false [](start = 92, length = 18)

Feels strange that the `true` comes first, despite corresponding with the last parameter. Am I reading this wrong?
Similarly, on F4, I don't understand the order of the flags (I would expect a `true` between two `false` flags)

Reply:
These are encoded in metadata order. Return parameter first, then actual parameters.

---
In reply to: [422433647](https://github.com/dotnet/roslyn/pull/44078#discussion_r422433647) [](ancestors = 422433647)